modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -1002,6 +1002,59 @@ TEST_P(Imgcodecs_Tiff_Modes, decode_multipage_use_memory_buffer)
     }
 }
 
+TEST_P(Imgcodecs_Tiff_Modes, decode_multipage_use_memory_buffer_selected_pages)
+{
+    const int mode = GetParam();
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filename = root + "readwrite/multipage.tif";
+    const string page_files[] = {
+        "readwrite/multipage_p1.tif",
+        "readwrite/multipage_p2.tif",
+        "readwrite/multipage_p3.tif",
+        "readwrite/multipage_p4.tif",
+        "readwrite/multipage_p5.tif",
+        "readwrite/multipage_p6.tif"
+    };
+    const size_t page_count = sizeof(page_files) / sizeof(page_files[0]);
+
+    FILE* fp = fopen(filename.c_str(), "rb");
+    ASSERT_TRUE(fp != NULL);
+    fseek(fp, 0, SEEK_END);
+    long pos = ftell(fp);
+
+    std::vector<uchar> buf;
+    buf.resize((size_t)pos);
+    fseek(fp, 0, SEEK_SET);
+    buf.resize(fread(&buf[0], 1, buf.size(), fp));
+    fclose(fp);
```

Comment:
```
size_t file_size = ftell(fp);

std::vector<uchar> buf(file_size);
fseek(fp, 0, SEEK_SET);
size_t actual_read = fread(&buf[0], 1, file_size, fp);
fclose(fp);
ASSERT_EQ(file_size, actual_read);
```

Reply:
If we are going to revise related code on the occasion (reading binary files would appear in both the tests: `decode_multipage_use_memory_buffer_{all,selected}_pages`), what about replacing:

    size_t file_size = ftell(fp);

    std::vector<uchar> buf(file_size);
    fseek(fp, 0, SEEK_SET);

with:

    size_t file_size = ftell(fp);
    fseek(fp, 0, SEEK_SET);

    std::vector<uchar> buf;
    buf.reserve(file_size);

In the proposed replacement, I/O is not interleaved by memory allocation and we do not unnecessarily initialize a chunk of memory that gets nearly immediately overwritten.